pandas/plotting/_matplotlib/core.py

```diff
@@ -969,6 +966,16 @@ def _make_plot(self):
         else:
             c_values = c
 
+        # plot a colorbar only if a colormap is provided or necessary
+        from matplotlib.colors import is_color_like
+
+        c_is_column_not_containing_colors = c_is_column and not all(
```

Comment:
maybe currently we should only consider if `c_values` are numeric, instead of checking if not all are color-like?

because your other PR are still under development, and seems no concrete conclusion on how we will show categorical/color-like in colorbar/legend, so now only having numeric values on colorbar seems most reasonable.

Reply:
Yes, good point - thanks for your review, @charlesdong1991 !